apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -375,20 +375,30 @@ const OutOfOfficePage = () => {
   }, [openModalOnStart]);
 
   const [openModal, setOpenModal] = useState(false);
+
+  const { data } = trpc.viewer.outOfOfficeReasonList.useQuery();
```

Comment:
Can we just extract the` isPending` status and use it instead of this: `const isBeforeFlicker = data?.length === undefined || data.length <= 0;`

Also, I found that this query is called from two different components. I think it would be better to pass the data and isPending status to the <OutOfOfficeEntriesList /> component instead of calling it two time.
Lmk what you think

Reply:
It seems `isPending` is enough, thanks for noticing.

![image](https://github.com/user-attachments/assets/4e6bc8ca-51eb-4d7d-8795-a050a54f0eca)

I don't think passing tRPC x React-Query data to a component that uses the same endpoint is the wisest decision because React-Query (even the default Next.js default fetch did this too) will handle the de-duplication using its Context. When we passed the value from the parent, the React Query Context and deduplication became useless. As you can see when we installed `react-query-devtools` the endpoint was just called once and distributed to all components using it.